runtime/queries/zig/highlights.scm

```diff
@@ -1,231 +1,294 @@
+  "resume"
+] @keyword
 
-(BreakLabel (IDENTIFIER) @label)
-(BlockLabel (IDENTIFIER) @label)
+"fn" @keyword.function
 
 [
-  "true"
-  "false"
-] @constant.builtin.boolean
+  "and"
+  "or"
+  "orelse"
+] @keyword.operator
 
 [
-  "undefined"
+  "try"
   "unreachable"
-  "null"
-] @constant.builtin
+  "return"
+] @keyword.control.return
 
 [
-  "else"
   "if"
+  "else"
   "switch"
+  "catch"
 ] @keyword.control.conditional
 
 [
   "for"
   "while"
+  "break"
+  "continue"
 ] @keyword.control.repeat
 
 [
-  "or"
-  "and"
-  "orelse"
-] @keyword.operator
-
-[
-  "enum"
-] @type.enum
+  "usingnamespace"
+  "export"
+] @keyword.control.import
 
 [
-  "struct"
-  "union"
-  "packed"
-  "opaque"
-  "export"
-  "extern"
-  "linksection"
-] @keyword.storage.type
+  "defer"
+  "errdefer"
+] @keyword.control.exception
 
 [
-  "const"
-  "var"
-  "threadlocal"
-  "allowzero"
   "volatile"
+  "allowzero"
+  "noalias"
+  "addrspace"
   "align"
+  "callconv"
+  "linksection"
+  "pub"
+  "inline"
+  "noinline"
+  "extern"
+  "comptime"
+  "packed"
+  "threadlocal"
 ] @keyword.storage.modifier
 
-[
-  "try"
-  "error"
-  "catch"
-] @keyword.control.exception
+; Builtins
 
 [
-  "fn"
-] @keyword.function
+  "c"
+  "..."
+] @variable.builtin
 
-[
-  "test"
-] @keyword
+((identifier) @variable.builtin
+  (#eq? @variable.builtin "_"))
 
-[
-  "pub"
-  "usingnamespace"
-] @keyword.control.import
+(calling_convention
+  (identifier) @variable.builtin)
 
-[
-  "return"
-  "break"
-  "continue"
-] @keyword.control.return
+; Modules
 
-[
-  "defer"
-  "errdefer"
-  "async"
-  "nosuspend"
-  "await"
-  "suspend"
-  "resume"
-] @function.macro
+(variable_declaration
+  (identifier) @variable ; TODO: module
+  (builtin_function
+    (builtin_identifier) @keyword.control.import
+    (#any-of? @keyword.control.import "@import" "@cImport")))
 
-[
-  "comptime"
-  "inline"
-  "noinline"
-  "asm"
-  "callconv"
-  "noalias"
-] @keyword.directive
+; Functions
 
-[
-  (CompareOp)
-  (BitwiseOp)
-  (BitShiftOp)
-  (AdditionOp)
-  (AssignOp)
-  (MultiplyOp)
-  (PrefixOp)
-  "*"
-  "**"
-  "->"
-  ".?"
-  ".*"
-  "?"
-] @operator
+(call_expression
+  function: (field_expression
+    member: (identifier) @function.method))
 
-[
-  ";"
-  "."
-  ","
-  ":"
-] @punctuation.delimiter
+(call_expression
+  function: (identifier) @function)
+
+(function_declaration
+  name: (identifier) @function)
+
+(builtin_identifier) @function.builtin
+
+; Fields
+
+(field_initializer
+  .
+  (identifier) @variable.other.member)
+
+(field_expression
+  (_)
+  member: (identifier) @variable.other.member)
+
+(container_field
+  name: (identifier) @variable.other.member)
+
+(initializer_list
+  (assignment_expression
+      left: (field_expression
+              .
+              member: (identifier) @variable.other.member)))
+
+; Labels
+
+(block_label (identifier) @label)
+
+(break_label (identifier) @label)
+
+; Constants
+
+((identifier) @constant
+  (#lua-match? @constant "^[A-Z][A-Z_0-9]+$"))
```

Comment:
Same here and below on L270 about `#lua-match?`

Reply:
```suggestion
  (#match? @constant "^[A-Z][A-Z_0-9]+$"))
```